### PR TITLE
migrate examples to openApi part 16; affects [gitlabrelease swagger]

### PR DESCRIPTION
**Pull request by chris48s**
Refs https://github.com/badges/shields/issues/9285

In this PR, I've started tackling query params.
Instead of converting a lot of services, I've concentrated on picking a couple of services that I think cover all of the most common variations so we can see what that looks like and establish the patterns we're going to use going forwards.

Can anyone think of a case I have missed here?

### CODE REVIEW

File: services/gitlab/gitlab-release.service.js [hunk 1]
```diff
@@ -38,58 +39,44 @@ export default class GitLabRelease extends GitLabBase {
...
-      ...commonProps,
-      queryParams: {
-        sort: 'semver',
-        include_prereleases: null,
+  static openApi = {
+    '/gitlab/v/release/{project}': {
+      get: {
+        summary: 'GitLab Release',
+        description: documentation,
+        parameters: [
```
Comment (chris48s) [inline services/gitlab/gitlab-release.service.js:47]: In situations where we've got a mix of path and query params, I think an array of calls to `pathParam()` and `queryParam()` is a bit nicer than concatenating the results of a call to `pathParams()` and a call to `queryParams()`

**Comment by chris48s** [comment]
Once we nail this one, I think we've got the helpers in place for most of the cases we need to deal with. Certainly the most common ones. We'll also have a fairly decent selection of examples to crib from in the codebase already. After this, I will look at updating the documentation to document these helpers and say that new services should declare an `openApi` property instead of `examples`.

**Comment by chris48s** [comment]
@calebcartwright @PyvesB Would it be possible for one of you to carve out a bit of time to have a look at this one? A bit of feedback on this particular PR would really help me to keep making progress on https://github.com/badges/shields/issues/9285

File: services/gitlab/gitlab-release.service.js [hunk 2]
```diff
@@ -38,58 +39,44 @@ export default class GitLabRelease extends GitLabBase {
...
+            example: 'semver',
+          }),
+          queryParam({
+            name: 'display_name',
+            schema: { type: 'string', enum: displayNameEnum },
+            example: 'release',
+          }),
+          queryParam({
+            name: 'date_order_by',
+            schema: { type: 'string', enum: dateOrderByEnum },
```
[the remark this answers is not in the document]
Comment (PyvesB) [reply services/gitlab/gitlab-release.service.js:73]: It's easier to follow things this way, let's stick to it.

[the remark this answers is not in the document]
Comment (PyvesB) [reply services/gitlab/gitlab-release.service.js:73]: It's easier to follow things this way, let's stick to it.

**Review by PyvesB — APPROVED** [review]
Looks good to me. Sorry for the delayed review. 👍🏻

**Comment by chris48s** [comment]
Thanks for having a look. I'll pick up next steps from here..

**Pull request merged**